File: powxn.cpp
#include <iostream>
using namespace std;

double myPow(double x, int n) {
  if (n==0) {
    return 1;
  }
  if (n >=1) {
    return x * myPow(x, n-1);
  }
  if (n <=-1)
  {
    return (1/x) * myPow(x, n+1);
  }
}

int main()
{
  double x = 2.0;
  cout << myPow(x, -2) << endl;
  cout << myPow(x, 2) << endl;
}

File: powxn.cs
using System;

public class Program
{
  public static int Main() {
    Console.WriteLine("Please e
    double x = Convert.ToDouble(Console.ReadLine());
    int n = Convert.ToInt32(Console.ReadLine());
    double res = myPow(x, n);
    Console.WriteLine(res);
    return 0;
  }
  public static double MyPow(double x, int n){
    if (n == 0){
      return 1.0;
    }
    else if (n>0)
    {
      return x * MyPow(x, n-1);
    }
    else
    {
      return (1/x) * MyPow(x, n-1);
    }
  }
}
